tests/test-multi-pow.c

```diff
@@ -0,0 +1,107 @@
+/* Test program for thread-safe dcurl */
+#include <pthread.h>
+#include "common.h"
+#include "dcurl.h"
+
+#define THREAD_MAX 10
+
+typedef struct _ducrl_item DCURL_ITEM;
+struct _ducrl_item {
+    int mwm;
+    int8_t input_trytes[TRANSACTION_TRYTES_LENGTH];  /* 2673 */
+    int8_t output_trytes[TRANSACTION_TRYTES_LENGTH]; /* 2673 */
+};
+
+void *dcurl_entry_cb(void *arg)
+{
+    DCURL_ITEM *dcurl_item = (DCURL_ITEM *) arg;
+    /* test dcurl Implementation with mwm = 9 */
+    int8_t *ret_trytes =
+        dcurl_entry(dcurl_item->input_trytes, dcurl_item->mwm, 0);
+    assert(ret_trytes);
```

Comment:
You can append messages with assert macro. e.g. `assert(a == b && "A is not equal to B");`